pkg/registry/core/service/storage/rest.go

```diff
@@ -374,12 +402,18 @@ func (rs *REST) Update(ctx context.Context, name string, objInfo rest.UpdatedObj
 		return nil, false, err
 	}
 
+	// set Service IP Family
+	if utilfeature.DefaultFeatureGate.Enabled(features.IPv6DualStack) && service.Spec.IPFamily == nil {
```

Comment:
I am pretty sure we don't need this any more, since defaulting will do this.  And in fact, will do it more correctly that this by looking at clusterIP.  What do you think?

Reply:
the reason this is here is for `pre existing services` on a cluster that just turned on `DualStack`. There is no other parts of the system that sets it for existing services. Along with that infamous `if statement for immutability` they ensure turning on `dualstack` on existing services will update existing services.

Reply:
An update of an existing service will go through defaulting, so the read will find a correct value.

Maybe I am missing something?

Reply:
this *is* the defaulting code. 

I feel this comment and https://github.com/kubernetes/kubernetes/pull/79386#discussion_r317324675 are related, let us close them on Monday.